src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -161,6 +161,15 @@ trait Symbols { this: Context =>
           owner.thisType, modcls, parents, decls, TermRef.withSymAndName(owner.thisType, module, name)),
         privateWithin, coord, assocFile)
 
+  def synthesizeCompanionMethod(name: TermName, ret: SymDenotation, owner: SymDenotation)(implicit ctx: Context) = {
+    if(owner.exists && ret.exists) ctx.newSymbol(
+      owner = owner.symbol,
+      name = name,
+      flags = Flags.Synthetic | Flags.Private,
+      info = ExprType(ret.typeRef))
+    else NoSymbol
+  }
+
   /** Create a package symbol with associated package class
```

Comment:
- Add a doc comment. 
- Formatting: if (.
- It's more common to pass owner as a symbol
- "ret" is not descriptive.
- In this case, I do not think the named parameters add anything. All arguments make it clear by their names already what parameter they are for.

I'd reformulate as follows:

    def synthesizeCompanionMethod(target: Symbol, owner: Symbol)(implicit ctx: Context) = {
       val name = if (target is Module) nme.COMPANION_MODULE_METHOD else nme.COMPANION_CLASS_METHOD
       ctx.newSymbol(owner, name, Flags.Synthetic | Flags.Private, ExprType(target.typeRef)
    }